pwndbg/aglib/kernel/vmmap.py

```diff
@@ -248,6 +247,84 @@ def kernel_vmmap_via_page_tables() -> Tuple[pwndbg.lib.memory.Page, ...]:
 monitor_info_mem_not_warned = True
 
 
+def _parser_mem_info_line_x86(line: str) -> pwndbg.lib.memory.Page | None:
+    """
+    Example response from `info mem`:
+    ```
+    ffff903580000000-ffff903580099000 0000000000099000 -rw
+    ffff903580099000-ffff90358009b000 0000000000002000 -r-
+    ffff90358009b000-ffff903582200000 0000000002165000 -rw
+    ffff903582200000-ffff903582803000 0000000000603000 -r-
+    ```
+    """
+
+    dash_idx = line.index("-")
+    space_idx = line.index(" ")
+    rspace_idx = line.rindex(" ")
+
+    start = int(line[:dash_idx], 16)
+    end = int(line[dash_idx + 1 : space_idx], 16)
+    size = int(line[space_idx + 1 : rspace_idx], 16)
+    perm = line[rspace_idx + 1 :]
+
+    flags = 0
+    if "r" in perm:
+        flags |= 4
+    if "w" in perm:
+        flags |= 2
+    if "x" in perm:
+        flags |= 1
+
+    global monitor_info_mem_not_warned
+    if end - start != size and monitor_info_mem_not_warned:
+        print(
+            M.warn(
+                (
+                    "The vmmap output may be incorrect as `monitor info mem` output assertion/assumption\n"
+                    "that end-start==size failed. The values are:\n"
+                    "end=%#x; start=%#x; size=%#x; end-start=%#x\n"
+                    "Note that this warning will not show up again in this Pwndbg/GDB session."
+                )
+                % (end, start, size, end - start)
+            )
+        )
+        monitor_info_mem_not_warned = False
+
+    return pwndbg.lib.memory.Page(start, size, flags, 0, "<qemu>")
```

Comment:
```suggestion
    return pwndbg.lib.memory.Page(start, size, flags, 0, "<qemu>")
```

Fwiw we should go ahead and do sth like `<mem_0xfff...>` where fff is few first bytes?

This may be helpful for commands that allow for filtering by page name (like search).

Though I am not sure since the addresses are big and this may just be super long?